src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -2773,6 +2781,22 @@ ALWAYS_INLINE Value ByteCodeInterpreter::decrementOperation(ExecutionState& stat
     }
 }
 
+NEVER_INLINE void ByteCodeInterpreter::ensureArgumentsObjectOperation(ExecutionState& state, ByteCodeBlock* byteCodeBlock, Value* registerFile)
+{
+    ExecutionState* es = &state;
+    while (es) {
+        if (es->lexicalEnvironment()->record()->isDeclarativeEnvironmentRecord() && es->lexicalEnvironment()->record()->asDeclarativeEnvironmentRecord()->isFunctionEnvironmentRecord()) {
+            break;
+        }
```

Comment:
What are the cases that this while loop is used for? Is there any case that ArgumentsObject is used here but it is allocated on the outer env record except the arrow function?

Reply:
For arrow function cases, it seems that ArgumentsObject is allocated for arrow function's env because arrow functions also have `FunctionEnvironmentRecord`.

Reply:
the loop is needed for try, with, block scope because it creates new env for it
and we are not use ensureargumentsobject opcode on arrow function
if there is arguments object reference on arrow function(access upper function object), I ensure arguments before arrow function expression not inside of arrow function